src/actions/VaccineActions.js

```diff
@@ -164,8 +163,13 @@ const scanForSensors = (dispatch, getState) => {
   BleService().scanForSensors(deviceCallback);
 };
 
-const startDownloadAllLogs = macAddress => async (dispatch, getState) => {
-  await withPermissions(dispatch, getState, downloadAllLogs(macAddress));
+const startDownloadAllLogs = () => async (dispatch, getState) => {
+  // Ensure there isn't already a download in progress before starting a new one
+  const state = getState();
+  const isDownloadingTemps = selectIsSyncingTemps(state);
```

Comment:
I was confused for a moment because the method is `DownloadAllLogs` and the condition is `DownloadingTemps` - it would help me if they were both the same.. either logs or temps

Reply:
Thanks - yeah that was a bit confusing, just keeping everyone on their toes 😆 . 
Have stuck with `DownloadAllLogs`, also renamed the selector to avoid confusion between Syncing and Downloading 3dc74637